src/main.rs

```diff
@@ -55,26 +60,38 @@ fn main() {
     let exercises = toml::from_str::<ExerciseList>(toml_str).unwrap().exercises;
 
     if let Some(ref matches) = matches.subcommand_matches("run") {
-        let filename = matches.value_of("file").unwrap_or_else(|| {
-            println!("Please supply a file name!");
+        let name = matches.value_of("name").unwrap_or_else(|| {
+            println!("Please supply an exercise name!");
             std::process::exit(1);
         });
 
-        let matching_exercise = |e: &&Exercise| {
-            Path::new(filename)
-                .canonicalize()
-                .map(|p| p.ends_with(&e.path))
-                .unwrap_or(false)
-        };
+        let matching_exercise = |e: &&Exercise| name == e.name;
 
         let exercise = exercises.iter().find(matching_exercise).unwrap_or_else(|| {
-            println!("No exercise found for your file name!");
+            println!("No exercise found for your given name!");
```

Comment:
Now that we rely on exercise names, I wonder whether there should be a test to check that exercise names are unique.

Reply:
Probably, but it'd be out of scope for this PR